src/DotLiquid.Tests/CustomIndexableTests.cs

```diff
@@ -70,15 +70,15 @@ public object ToLiquid()
         public void TestVirtualListLoop()
         {
             string output = Template.Parse("{%for item in list%}{{ item }} {%endfor%}")
-                .Render(Hash.FromAnonymousObject(new {list = new VirtualList(1, "Second", 3)}));
+                .Render(Hash.FromAnonymousObject(new {list = new VirtualList(1L, "Second", 3L)}));
             Assert.AreEqual("1 Second 3 ", output);
         }
 
         [Test]
         public void TestVirtualListIndex()
         {
             string output = Template.Parse("1: {{ list[0] }}, 2: {{ list[1] }}, 3: {{ list[2] }}")
-                .Render(Hash.FromAnonymousObject(new {list = new VirtualList(1, "Second", 3)}));
+                .Render(Hash.FromAnonymousObject(new {list = new VirtualList(1L, "Second", 3L)}));
```

Comment:
list index should at minimum keep supporting int, TBD if long makes any sense here.